src/query/procedure/mg_procedure_impl.cpp

```diff
@@ -2109,7 +2295,18 @@ mgp_error mgp_graph_delete_vertex(struct mgp_graph *graph, mgp_vertex *vertex) {
     if (!MgpGraphIsMutable(*graph)) {
       throw ImmutableObjectException{"Cannot remove a vertex from an immutable graph!"};
     }
-    const auto result = graph->impl->RemoveVertex(&vertex->impl);
+
+    const auto result =
+        std::visit(memgraph::utils::Overloaded{
+                       [&](memgraph::query::DbAccessor *impl) {
+                         MG_ASSERT(std::holds_alternative<memgraph::query::VertexAccessor>(vertex->impl));
+                         return impl->RemoveVertex(&std::get<memgraph::query::VertexAccessor>(vertex->impl));
+                       },
+                       [&](memgraph::query::SubgraphDbAccessor *impl) {
+                         MG_ASSERT(std::holds_alternative<memgraph::query::SubgraphVertexAccessor>(vertex->impl));
+                         return impl->RemoveVertex(&(std::get<memgraph::query::SubgraphVertexAccessor>(vertex->impl)));
```

Comment:
Use one `lambda` with `auto*`. Remove `Overloaded`

Reply:
I think I can't use auto* since in one case I am sending VertexAccessor and in other SubgraphVertexAccessor. And also I am doing MG_ASSERT on correct type for SubgraphDbAccessor so that inside that function I have SubgraphVertexAccessor.

Reply:
Same comment on the rest :)

Reply:
I think we resolved this.